### PR TITLE
Core bug fixes

### PR DESCRIPTION
**Pull request by kay-jahnke**
- Making Error messages sticky: The Error messages now stay on top but they do not need to be focused (non-modal).
- deleting the restart method in the manager and rather making use of the realQuit, because only this one respects the dependencies

## Description

When clicking the restart Button in the manager qudi was restarted, but the individual modules were deloaded in "random" order. This lead to cases where the hardware module was deactivated while the logic still tried to access it. The realQuit does respect the dependencies, so it was easy to just extend it to also handle the restart. the old restart method was then deleted.

Also ran an automatic code cleaning on the files I touched.

## Motivation and Context

- Error messages got lost when too many windows were open on the computer.
- Restarts of qudi produced unnecessary errors that might also damage hardware

## How Has This Been Tested?

tested with dummy config(s) on Linux 5.7.9-1-MANJARO

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have documented my changes in the changelog (`documentation/changelog.md`)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added/updated for the module the config example in the docstring of the class accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [x] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

File: core/manager.py [hunk 1]
```diff
@@ -1154,6 +1157,17 @@ def startModule(self, base, key):
             If the module is already loaded, just activate it.
             If the module is an active GUI module, show its window.
         """
+
+        if QtCore.QThread.currentThread() is not self.thread():
+            QtCore.QMetaObject.invokeMethod(
```
Comment (Neverhorst) [inline core/manager.py:1162]: Might this cause the trouble I mentioned?

**Review by Neverhorst — COMMENTED** [review]
While testing with default config, qudi hangs after opening the manager window (probably while automatically loading modules for tasklogic) and does not respond anymore. I suspect the lines 1160ff in _manager.py_.

File: core/manager.py (hunk 1 above)
Comment (kay-jahnke) [reply core/manager.py:1162]: Thanks for the tests, I think I will need to double check on Windows. I will then come back to you.

Comment (kay-jahnke) [reply core/manager.py]: I deleted this part of the pull request because it is not easily solvable in the old core. New core handles it automatically.

**Review by Neverhorst — APPROVED** [review]
Seems to work now. Shall we merge?

**Pull request merged**